repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/oauth/OAuthProvider.java

```diff
@@ -24,10 +30,52 @@
   public OAuthProvider(String id, String label, IOAuthFlowConfiguration configuration) {
     this.id = id;
     this.label = label;
-    this.logoHref = configuration.getLogoHref();
-    this.loginUrl = configuration.getFilterProcessingUrl();
+    this.logoHref = configuration != null ? configuration.getLogoHref() : "";
+    this.loginUrl = configuration != null ? configuration.getFilterProcessingUrl() : "";
   }
-  
+
+  /**
+   * The only purpose of this ugly utility method is to "merge" the various versions of Bosch IoT
+   * Suite OAuth provider (v.1 and v.2 at the time of writing), in order to present a single
+   * provider to the end user, who is agnostic to which version they will be using. <br/>
+   * This feature is used in the {@link org.eclipse.vorto.repository.web.HomeController}, when
+   * populating the list of non-webflow {@link IOAuthProvider} DTOs for the UI. <br/>
+   * In the UI, this is not used at login (since we are dealing with non-webflow providers), but
+   * rather when choosing the desired authentication methodology for new technical users.<br/>
+   * Note: the merging is done by comparing the label to a constant, since importing the suite oauth
+   * providers would introduce a circular dependency.
+   */
+  public static List<OAuthProvider> mergeBoschIotSuiteOAuthProviders(List<OAuthProvider> providers) {
+    // reducing any number of providers with Bosch Suite OAuth label to one, if applicable
+    // we don't care which one, as both versions have the same presentational properties ("fake" picture
+    // and label)
+    Optional<OAuthProvider> singleSuiteOAuthProvider = providers.stream().filter(i -> i.getLabel().equals(BOSCH_SUITE_OAUTH_LABEL)).reduce((a,b) -> a);
+    // getting non-Bosch Suite OAuth providers if any
+    List<OAuthProvider> result = providers.stream().filter(i -> !i.getLabel().equals(BOSCH_SUITE_OAUTH_LABEL)).collect(
+        Collectors.toList());
+    // adding reduced Suite OAuth providers if any
+    if (singleSuiteOAuthProvider.isPresent()) {
+      result.add(singleSuiteOAuthProvider.get());
+    }
+    return result;
+  }
+
+  public static OAuthProvider from(IOAuthProvider source) {
+    OAuthProvider result = new OAuthProvider();
+    result.setId(source.getId());
+    result.setLabel(source.getLabel());
+    if (source.getWebflowConfiguration().isPresent()) {
+      IOAuthFlowConfiguration config = source.getWebflowConfiguration().get();
+      result.setLogoHref(config.getLogoHref());
+      result.setLoginUrl(config.getFilterProcessingUrl());
+      // TODO this is temporary and ugly: assigning the Bosch ID icon to the non-webflow providers
```

Comment:
TODO? What's left to do here?

Reply:
> TODO? What's left to do here?

Resolved conversation by mistake. I think the TODO should stay until we figure out what exactly we want to do with the icon.

Reply:
tbh I don't think we will ever do smth about the icon.

Reply:
@kolotu ye of little faith. Allright I'll bury the TODO then

Reply:
@mena-bosch thanks :)